api/src/opentrons/protocol_runner/task_queue.py

```diff
@@ -27,40 +28,35 @@ class TaskQueue:
 
     def __init__(self) -> None:
         """Initialize the TaskQueue."""
-        self._run_entry: Optional[TaskQueueEntry] = None
-        self._cleanup_entry: Optional[TaskQueueEntry] = None
+        self._run_func: Optional[Callable[[], Any]] = None
+        self._cleanup_func: Optional[CleanupFunc] = None
         self._run_task: Optional["asyncio.Task[None]"] = None
         self._run_started_event: asyncio.Event = asyncio.Event()
 
-    def add(
+    def add_run_func(
         self,
-        phase: TaskQueuePhase,
         func: Callable[..., Awaitable[Any]],
-        *args: Any,
         **kwargs: Any,
     ) -> None:
-        """Add a task to the queue.
+        """Add the protocol run task to the queue.
 
-        Two phases are available: TaskQueuePhase.RUN and TaskQueuePhase.CLEANUP.
-        Each phase may contain no tasks or one task. The RUN task will be run first,
-        if present, and the CLEANUP task will run second. The CLEANUP task will run
-        regardless of the success or failure of the RUN task.
+        The "run" task will be run first, before the "cleanup" task.
         """
-        entry = TaskQueueEntry(func=func, args=args, kwargs=kwargs)
+        self._run_func = partial(func, **kwargs)
 
-        if phase == TaskQueuePhase.RUN:
-            self._run_entry = entry
-        else:
-            self._cleanup_entry = entry
+    def add_cleanup_func(self, func: CleanupFunc) -> None:
+        """Add the run cleanup task to the queue.
 
-    def is_started(self) -> bool:
-        """Get whether the task queue has started running."""
-        return self._run_task is not None
+        The "cleanup" task will run after the "run" task, and will be passed
+        any exceptions raised by the "run" task.
+        """
+        self._cleanup_func = func
 
     def start(self) -> None:
         """Start running tasks in the queue."""
-        self._run_task = asyncio.create_task(self._run())
-        self._run_started_event.set()
+        if self._run_task is None:
```

Comment:
Should this be:

```suggestion
        if not self._run_started_event.is_set():
```

I think it's effectively equivalent, but using `self._run_started_event` avoids creating a competing/redundant source of truth.

Reply:
On second thought, `.join()` already uses `if self._run_task`. So I guess this isn't a new problem. Feel free to ignore.